pkg/client/metadata/metadata.go

```diff
@@ -53,7 +54,11 @@ func GetDynatraceClassicURL(ctx context.Context, client *rest.Client, environmen
 
 	resp, err := client.Get(ctx, endpointURL)
 	if !resp.IsSuccess() || err != nil {
-		log.Debug("failed to query classic environment url from %q, falling back to deprecated endpoint %q: %v (HTTP %v)", ClassicEnvironmentDomainPath, DeprecatedClassicEnvDomainPath, err, resp.StatusCode)
+		if err == nil {
+			log.Debug("failed to query classic environment url from %q, falling back to deprecated endpoint %q (HTTP %v)", ClassicEnvironmentDomainPath, DeprecatedClassicEnvDomainPath, resp.StatusCode)
```

Comment:
I think it'd be helpful to also log the response body if this happens. The status code alone might not tell a user or us enough to resolve an issue with this